presto-parquet/src/main/java/io/prestosql/parquet/writer/valuewriter/DecimalValueWriter.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.parquet.writer.valuewriter;
+
+import io.airlift.slice.Slice;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.type.DecimalType;
+import io.prestosql.spi.type.Type;
+import org.apache.parquet.column.statistics.Statistics;
+import org.apache.parquet.column.values.ValuesWriter;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.PrimitiveType;
+
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+public class DecimalValueWriter
+        extends PrimitiveValueWriter
+{
+    private final BiConsumer<Block, Integer> writeBlockAt;
+
+    public DecimalValueWriter(ValuesWriter valuesWriter, Type type, PrimitiveType parquetType)
+    {
+        super(parquetType, valuesWriter);
+        this.writeBlockAt = getWriteFunction(valuesWriter, type, this::getStatistics);
+    }
+
+    @Override
+    public void write(Block block)
+    {
+        for (int i = 0; i < block.getPositionCount(); ++i) {
+            if (!block.isNull(i)) {
+                writeBlockAt.accept(block, i);
+            }
+        }
+    }
+
+    private static BiConsumer<Block, Integer> getWriteFunction(ValuesWriter valuesWriter, Type type, Supplier<Statistics<?>> statisticsSupplier)
```

Comment:
I would move the if statements to `write` and have three loops.  It is faster to branch on the outside of the loop.

Reply:
Got it. That makes more sense.